File: 003/largestPrime.scala
val num = 600851475143L 

def factors(n: Long): List[Long] = 
  (2 to math.sqrt(n).toInt).find(n % _ == 0)
    .map(i => i.toLong :: factors(n / i)).getOrElse(List(n))

println(factors(num).last)

File: 003/largest_prime.py
def prime_factors(n):
	factors = list()
	i = 2
	while i <= n:
		while n % i == 0:
			n /= i
			factors.append(i)
		i += 1
	return factors
	
val = 600851475143

print(prime_factors(val))
